### ISSUE TITLE
400: No permission to add media items to this album., badRequest

### ISSUE THREAD

**Issue by wormiedk** [gphotosuploader/gphotos-uploader-cli#49]
When I try to upload my photos I get this error:
--- at /Users/david/go/src/github.com/nmrshll/google-photos-api-client-go/lib-gphotos/client.go:115 (Client.UploadFile) ---
Caused by: googleapi: Error 400: No permission to add media items to this album., badRequest

Before this error was thrown gphotos-uploader-cli was able to upload quite a few files to the given album

**Comment by ndom91** [comment]
I am having the same issue - my initial upload worked great, according to the dev console I made approximately 1,900 requests - but now I can't upload anything further. This is a business G Suite account so I shouldn't have any capacity limits, and even days later I cant upload any more so any sort of daily limits should have been reset by now. 

Anyone have any idea on how to further debug this, or get a more detailed error message?

**Comment by pacoorozco** [comment]
can you test our last release?

@wormiedk @ndom91

**Comment by ndom91** [comment]
@pacoorozco unfortunately I dont work at this company anymore and dont have access to the server. Best of luck though!

**Comment by wormiedk** [comment]
Testing at the moment. It seems to work. However the upload initially failed because I forgot to omit .DS_STORE files and ran into an error for those. This seems to have generated empty albums.

**Comment by HydrelioxGitHub** [comment]
Can I re-open this issue ? I've the same error but for me it never worker. I've a test batch with some pictures and each one got this error.

```
root@nas:~# docker exec -it gphoto run                                                                                                                                                                             
[info]   Found item to upload: /photos/Voici Gohan/2015-03-17.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104822.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104829.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104831.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104840.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104841.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104844.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104849.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_104909.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_105556.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_105613.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_110417.jpg
[info]   Found item to upload: /photos/Voici Gohan/IMG_20150315_110445.jpg
[info]   Found item to upload: /photos/Voici Gohan/VID_20150315_110420.mp4
[info]   14 files pending to be uploaded in folder '/photos'.
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104840.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104840.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104841.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104841.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104844.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104844.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104849.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104849.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104909.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104909.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_105613.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_105613.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_105556.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_105556.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_110417.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_110417.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_110445.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_110445.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104822.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104822.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/2015-03-17.jpg, err=googleapi: Error 400: No permission to add media items to this album., badReques
t                                                                                                                                                                                                                  
[fail] X Error processing /photos/Voici Gohan/2015-03-17.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104829.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104829.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/IMG_20150315_104831.jpg, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/IMG_20150315_104831.jpg
[error]  error while trying to create this media item, err=unexpected error response: file=/photos/Voici Gohan/VID_20150315_110420.mp4, err=googleapi: Error 400: No permission to add media items to this album., 
badRequest                                                                                                                                                                                                         
[fail] X Error processing /photos/Voici Gohan/VID_20150315_110420.mp4
[info]   14 processed files: 0 successfully, 14 with errors
```

**Comment by pacoorozco** [comment]
Hi @HydrelioxGitHub 

Can you give me the version of the binary that you are using? Can you confirm if the folder `Voici Gohan` exists on Google Photos? 

In case it exists, can you rename it and try to upload again?

Paco

**Comment by HydrelioxGitHub** [comment]
I'm using the dockerized version from @rfgamaral . The container is using v. 1.0.4

```
root@nas:~# docker exec -it gphoto /bin/sh                                                                                                  
                                                 
/ # gphotos-uploader-cli version                                                                                                            
gphotos-uploader-cli 1.0.4-docker/ #    
```
I've tried to rename or even delete the album but the error still the same.

**Comment by HydrelioxGitHub** [comment]
I've tried to just change the name of the folder as a random name ( plop1234 ) with the same pictures inside and try to run the upload and it's works flawlessly. So the problem is really related to the album creation and not the medias itself.
I don't know if it helps ...

**Comment by pacoorozco** [comment]
Hi

Was this album created by gphotosuploader? I guess that out was not. If you
create the album before then the API has no permission over it:
https://developers.google.com/photos/library/guides/manage-albums#adding-items-to-album

Paco

**Comment by HydrelioxGitHub** [comment]
I understand the limitation. I created an album with that name, deleted it after. Should I be able to create it with the cli with that specific name ? 
Can we image an option for the cli that create an "albumname_1" if the "albumname" is not available ?

**Comment by HydrelioxGitHub** [comment]
I've made one test with some pictures and deleted them on Google Photo. Now these photo are not detected again as uploadable. How do I reset to re-upload them ?

**Comment by pacoorozco** [comment]
Hi,
Already uploaded files information is kept on `~/.config/gphotos-uploader-cli/uploads.db` (by default). If you remove this folder, `gphotos-uploader-cli` will lose these already uploaded files information.

**Comment by HydrelioxGitHub** [comment]
I've figured out something weird. I use this project on my desktop. And the album that caused the error was not in Google Photo when I searched for it on the web version of Google Photo. 
Today, I don't know why, I've searched for this problematic album through the android app and I've found an empty album with the same name.
I deleted it and retry the gphotos-uploader-cli and no error !

I don't why the search results does not give the same result on the two different platform but it explain the error.

However, thanks a lot for your work and your time guys.

**Comment by eng3** [comment]
I am just starting.  I already have 99% of my library on google photos (back in the day with Picasa and through manual album creation).  I tried letting the upload go on its own (failing every file) thinking the upload database will eventually become up to date and any new albums should get created moving forward.  I stopped gphotos because I reached my upload limit.  Upon restarting, I now get "uploads.db, err=resource temporarily unavailable".  I guess the uploads.db got corrupted?  What is the recommended way to get around this issue and the general issue of already having a library on google photos.  I'd prefer not to completely delete all my uploaded albums since I'm not 100% sure this tool even works for me.
